packages/new-polymath-issuer/src/pages/DividendsWizard/Step-2/index.tsx

```diff
@@ -116,14 +117,39 @@ export const Step2: FC<Props> = ({
     updateTaxWithholdingList(formattedValues);
   };
 
-  const handleValidation = (values: FormValues) => {
+  const handleValidation = async (values: FormValues) => {
+    const walletAddress = values.currentTaxWithholding[csvEthAddressKey];
+    let errors = {};
+
     try {
-      validateYupSchema(values, schema, true);
+      await validateYupSchema(values, schema, true);
     } catch (err) {
-      const errors = yupToFormErrors(err);
-      return errors;
+      errors = { ...errors, ...yupToFormErrors(err) };
     }
+
+    // Make sure wallet is existing and whitelisted
+    if (
```

Comment:
This is me being annoying but this part of the code is super hard to read, can we move these to more descriptive booleans? Example:

```ts
if (walletAddress && taxWithholdingIsValid && isNotExcluded) {
  // ...
}
```